tests/test_helper.tcl

```diff
@@ -15,21 +15,30 @@ source tests/support/util.tcl
 
 set dir [pwd]
 set ::all_tests []
+set ::module_api_all_tests []
 
 set test_dirs {
     unit
     unit/type
     unit/cluster
     integration
+    unit/moduleapi
 }
 
 foreach test_dir $test_dirs {
     set files [glob -nocomplain $dir/tests/$test_dir/*.tcl]
 
-    foreach file $files {
-        lappend ::all_tests $test_dir/[file root [file tail $file]]
+    if {$test_dir eq "unit/moduleapi"} {
+        foreach file $files {
+            lappend ::module_api_all_tests $test_dir/[file root [file tail $file]]
+        }
+    } else {
```

Comment:
Shall we just loop in a separate step rather than introducing this conditional statement? WDYT

```
foreach file [glob -nocomplain $dir/tests/unit/moduleapi] {
   lappend ::module_api_all_tests $test_dir/[file root [file tail $file]]
}
```

Reply:
this is actually my first version, i was afraid that there would be more module api dir in the future, but now it seems unnecessary. Yean, i am changing it!